libraries/AP_WayBack/AP_WayBack.cpp

```diff
@@ -0,0 +1,818 @@
+/*
+
+this library collects points of drone position, records them in memory and eliminates 
+unnecessary data - redundant points and loops, so it always knows 
+the shortest way to home via visited points.
+ 
+ Copyright night_ghost@ykoctpa.ru 2017
+ 
+ license GPLV3 or above
+   
+*/
+
+#include "AP_WayBack.h"
+
+extern const AP_HAL::HAL& hal;
+
+AP_WayBack::Point* AP_WayBack::points;
+uint16_t          AP_WayBack::max_num_points=0; // size of storage
```

Comment:
These variables are static members because this class is intended to be static by design: we have just only one point-of-start. Creation of another instances can be checked and prohibited if it needs